doc/type_requirements/File.qbk

```diff
@@ -0,0 +1,183 @@
+[/
+    Copyright (c) 2013-2016 Vinnie Falco (vinnie dot falco at gmail dot com)
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+]
+
+[section:File File]
+
+The [*File] concept abstracts access to files in the underlying file system.
+
+To support interfaces other than Win32 or POSIX, callers may provide their
+own [*File] type that meets these requirements.
+
+[heading Associated Types]
+* [link nudb.ref.nudb__path_type path_type]
+* [link nudb.ref.nudb__file_mode file_mode]
+* [link nudb.ref.nudb__error_code error_code]
+
+[heading Notation]
+
+* `F` a [*File] type
+* `f` an instance of type F
+* `p` an instance of type [link nudb.ref.nudb__path_type path_type]
+* `m` an instance of the type [link nudb.ref.nudb__file_mode file_mode]
+* `ec` an instance of a reference to the type
+      [link nudb.ref.nudb__error_code error_code]
+* `n` a number of bytes. Type is `std::size_t`
+* 'o' an offset from the beginning of a file. Type is `std::size_t`
+* 'a' a memory address of type void *
+
+[heading Valid Expressions]
+[table Valid Expressions
+[[Operation] [Return Type] [Semantics, Pre/Post-conditions]]
+[
+    [`F()`]
+    [ ]
+    [
+        Default constructable
+    ]
+]
+[
+    [`F(f)`]
+    [ ]
+    [
+        Move constructable.  Note NOT copy constructable.
+    ]
+]
+
+[
+    [`f.is_open()`]
```

Comment:
It's been custom to use the first letter as sort of an abbrevation for the thing it refers to.  To make this point I went to the canonical example of concepts (File is a concept) for an example: https://www.sgi.com/tech/stl/Container.html  Turns out I was wrong and that at least in this case it's not the custom the letter used to short hand  the type/object.  Sooo - there was no intended meaning from using the letter 'c' - not "const" nor anything else.  I still like my "custom/convention" but I can't argue that it's anything other my personal preference.

As an aside - does the File concept require that f.is_open() be const?  Since the File concepts describes things in terms of valid expressions, it can't make a such a requirement.  The only time is_open() being "const" would make a difference would be if a non-const member function called "is_open()".  But since File concept doesn't permit users to specify members of File, this error would never be detected by the user so it's not relevant anyway.  Of course in the implementation of the File concept, it's relevant but not to the usage of types implementing the file concept.  The whole idea of a "concept" is to decouple required behavior from the implemenation.